### PR TITLE
EQL: Remove "wildcard" function

### PR DESCRIPTION
**Pull request by bpintea**
This removes "wildcard" as an available function in EQL. This has
already been replace with "like" and "regex" embedded synthax (and
respective case insensitive variants).

Relates #71906.

### CODE REVIEW

File: x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml [hunk 1]
```diff
@@ -273,20 +273,6 @@ file where file_name : ("winini?.exe", "lsass.e?e") and opcode == 2
 '''
 expected_event_ids  = [65, 86]
 
-[[queries]]
-name = "wildcardFunctionWildcardPattern"
-query = '''
-file where wildcard(file_name, "winini*.exe", "lsass.*") and opcode == 2
```
Comment (rw-access) [inline x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml:279]: should this be replaced with `like`?

File: x-pack/plugin/eql/src/test/resources/test_string_functions.toml [hunk 2]
```diff
@@ -411,73 +411,3 @@ description = "Test the `substring` function when the case already matches"
     [[substring.fold.tests]]
     expression = '''substring("hello world", -5, -1)'''
     expected = "worl"
-
-[wildcard]
```
Comment (rw-access) [inline x-pack/plugin/eql/src/test/resources/test_string_functions.toml:415]: should this one be replaced with `like`? is this file even used?

**Review by rw-access — APPROVED** [review]
LGTM.
My only note is that I'm not sure where `wildcard` should be deleted and where it should be changed to `like`.

**Review by Luegg — APPROVED** [review]
Lgtm 👍

File: x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml (hunk 1 above)
Comment (bpintea) [reply x-pack/plugin/eql/qa/common/src/main/resources/additional_test_queries.toml:279]: There's already a `like` test for this (`likeMultipleArgWithPattern`).

File: x-pack/plugin/eql/src/test/resources/test_string_functions.toml (hunk 2 above)
Comment (bpintea) [reply x-pack/plugin/eql/src/test/resources/test_string_functions.toml:415]: Yes, file still in use.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryTranslatorFailTests.java [hunk 3]
```diff
@@ -205,35 +205,4 @@ public void testStringContainsWrongParams() {
         assertEquals("1:15: second argument of [stringContains(process_name, 1)] must be [string], found value [1] type [integer]",
                 error("process where stringContains(process_name, 1)"));
     }
-
-    public void testWildcardNotEnoughArguments() {
```
Comment (costin) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryTranslatorFailTests.java:209]: Please convert these to `like` if there aren't tests that already do that.

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/Wildcard.java [hunk 4]
```diff
@@ -31,7 +31,7 @@
 
 /**
  * EQL wildcard function. Matches the form:
- *     wildcard(field, "*wildcard*pattern*", ...)
+ *     field like ("*wildcard*pattern*", ...)
  */
 public class Wildcard extends BaseSurrogateFunction {
```
Comment (costin) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/Wildcard.java:36]: No need to keep this class around.
It's used inside the parser `ExpressionBuilder#visitOperatorExpressionDefault` however its only role is to call `makeSubstitute` which is similar to `ExpressionBuilder#combineExpressions`. See the handling of `EqlBaseParser.IN_INSENSITIVE:` in `visitOperatorExpressionDefault`.

**Review by costin — CHANGES_REQUESTED** [review]
Assuming the removed tests already have equivalent for `like` the only comment I have is around fully removing the `Wildcard` class which is not needed and the substitute construct can be created directly by the parser.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryTranslatorFailTests.java (hunk 3 above)
Comment (bpintea) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryTranslatorFailTests.java:209]: These tests become now syntactic failures. One case could however be indeed converted (ex `testWildcardWithNumericField`).

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/Wildcard.java (hunk 4 above)
Comment (bpintea) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/Wildcard.java]: Thanks. Class removed.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryTranslatorFailTests.java (hunk 3 above)
Comment (costin) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryTranslatorFailTests.java:209]: As long as we have equivalent semantical tests, I'm 👍 . Note that wildcard tests that rely on the syntax can be changed to `like/like~`  hence my point of replacing removed tests (or checking) there are semantical equivalents to them.

**Review by costin — APPROVED** [review]
LGTM

**Comment by bpintea** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by bpintea** [comment]
@elasticmachine update branch

**Pull request merged**